README.markdown
About
=====
20Couch - A Google Reader like Twitter Client

[20Couch.com](http://www.20couch.com)

By Matt Curry <matt@pseudocoder.com>

Install
=======
* Download and unpack the latest [CakePHP 1.3.x](http://github.com/cakephp/cakephp/downloads)
* Download and unpack the latest [20Couch](http://github.com/mcurry/20couch/downloads)
* Copy the contents of 20Couch into the CakePHP folder.  This will replace the entire app dir.
* Create a database for 20Couch on your web server, as well as a MySQL user who has all privileges for accessing and modifying it.
* Create an empty file "install" in /app/tmp.  This will allow the installer to run.
* Place the 20Couch and CakePHP files in a desirable location on your web server.  Other apps should be jealous of this location.
* Install by going to the 20Couch installer in your least favorite browser.
	* If you installed 20Couch in the root directory, you should visit: http://20couch.example.com/install
	* If you installed 20Couch in its own subdirectory called "unicorns" (for example) you should visit: http://example.com/unicorns/install
	
Optional
========
* Add a cron entry to run the updated behind the scenes (* * * * * sh /path/to/20couch/20couch-update.sh)
* Set your webroot to /path/to/20couch/app/webroot instead of just /path/to/20couch

Requirements
============
* Apache
* MySQL
* PHP 4 or 5


License
=======
Copyright (c) 2010 Matt Curry

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.

RELEASES.markdown
Version 1.12 (October 29, 2010)
=============================
Features
--------
* New mobile interface built on [jQuery Mobile](http://jquerymobile.com).

Version 1.11 (June 05, 2010)
=============================
Features
--------
* Track Replies - See all the replies to a tweet

Defects
-------
* #51 [FR: Default account on tags](http://20couch.lighthouseapp.com/projects/46779/tickets/51-fr-default-account-on-tags)
* #69 [Right click anywhere in a tweet toggles the actions](http://20couch.lighthouseapp.com/projects/46779/tickets/69-right-click-anywhere-in-a-tweet-toggles-the-actions)
* #75 [Long click to hide options is triggering Click Mark as Read plugin](http://20couch.lighthouseapp.com/projects/46779/tickets/75-long-click-to-hide-options-is-triggering-click-mark-as-read-plugin)
* #77 [Unfollow doesn't work](http://20couch.lighthouseapp.com/projects/46779/tickets/77-unfollow-doesnt-work)

Version 1.10 (May 05, 2010)
=============================
Features
--------
* General mobile improvements

Defects
-------
* #29 [Mobile version doesn't refresh](http://20couch.lighthouseapp.com/projects/46779/tickets/29-mobile-version-doesnt-refresh)
* #47 [Reply from mobile](http://20couch.lighthouseapp.com/projects/46779/tickets/47-reply-from-mobile)
* #68 ["Mark all as read" doesn't work](http://20couch.lighthouseapp.com/projects/46779/tickets/68-mark-all-as-read-doesnt-work)


Version 1.09 (April 23, 2010)
=============================
Features
--------
* Moved short url expando to plugin.  Better tooltip.
* Added click mark as read plugin.

Defects
-------
* #62 [Plugins do not register](http://20couch.lighthouseapp.com/projects/46779/tickets/62-plugins-do-not-register)
* #71 [DM has no origin](http://20couch.lighthouseapp.com/projects/46779/tickets/71-dm-has-no-origin)
* #72 [Navigation plugin does not recognise new items](http://20couch.lighthouseapp.com/projects/46779/tickets/72-navigation-plugin-does-not-recognise-new-items)

Version 1.08 (April 16, 2010)
=============================
Features
--------
* #1 - [Keyboard navigation](http://20couch.lighthouseapp.com/projects/46779/tickets/1-keyboard-navigation)

Defects
-------
* Fixed issue with auto update and exec(unzip).

Version 1.07 (April 15, 2010)
=============================
Features
--------
* #16 - [FR: Suggested completion for @<user>](http://20couch.lighthouseapp.com/projects/46779/tickets/16-fr-suggested-completion-for-user)

Defects
-------
* #70 - [Mobile: 'Mark all as read' button marks all accounts as read](http://20couch.lighthouseapp.com/projects/46779/tickets/70-mobile-mark-as-read-button-marks-all-accounts-as-read)

Version 1.06 (April 14, 2010)
=============================
Features
--------
* #55 - [FR: Ability to mark individual or blocks of posts as read](http://20couch.lighthouseapp.com/projects/46779/tickets/55-fr-ability-to-mark-individual-or-blocks-of-posts-as-read)

Defects
-------
* #61 - [Settings show up as text fields](http://20couch.lighthouseapp.com/projects/46779/tickets/61-settings-show-up-as-text-fields)
* #62 - [Plugins do not register](http://20couch.lighthouseapp.com/projects/46779/tickets/62-plugins-do-not-register)
* #66 - [Retweet doesn't update chracter count](http://20couch.lighthouseapp.com/projects/46779/tickets/66-retweet-doesnt-update-chracter-count)
* #67 - [Settings window lost its pop-ups](http://20couch.lighthouseapp.com/projects/46779/tickets/67-settings-window-lost-its-pop-ups)

Version 1.05 (April 13, 2010)
=============================
Features
--------
* #57 - [Update to Cake 1.3 RC4](http://20couch.lighthouseapp.com/projects/46779/tickets/57-update-to-cake-13)
* #58 - [Initial Plugin Support](http://20couch.lighthouseapp.com/projects/46779/tickets/58-initial-plugin-support)
* #60 - [iPad support](http://20couch.lighthouseapp.com/projects/46779/tickets/60-ipad-support)
* 57x57 icon for iPad

Defects
-------
* #56 - [Bug w/ drop down not appearing in the right place](http://20couch.lighthouseapp.com/projects/46779/tickets/56-split-button-for-mark-as-read#ticket-56-3)

Version 1.04 (April 12, 2010)
=============================
Features
--------
* #56 - [Split button for Mark As Read](http://20couch.lighthouseapp.com/projects/46779/tickets/56-split-button-for-mark-as-read)

Defects
-------
* None!

Version 1.03 (April 04, 2010)
=============================
Features
--------
* #52 - [Re-design threaded tweets](http://20couch.lighthouseapp.com/projects/46779/tickets/52-re-design-threaded-tweets)

Defects
-------
* #2 - [All items should have non obtrusive indication of origin account](http://20couch.lighthouseapp.com/projects/46779/tickets/2-all-items-should-have-non-obtrusive-indication-of-origin-account)
* #41 - [FR: Add quick and dirty Un-Follow link.](http://20couch.lighthouseapp.com/projects/46779/tickets/41-fr-add-quick-and-dirty-un-follow-link)
* #50 - [Duplicate items](http://20couch.lighthouseapp.com/projects/46779/tickets/50-duplicate-items)

Version 1.02 (March 24, 2010)
=============================
Features
--------
* Updated jQuery UI to 1.8.0

Defects
-------
* Fixed a few IE issues
* #33 - [Send a tweet form needs to refresh many of it's UI elements and values](http://20couch.lighthouseapp.com/projects/46779/tickets/33-send-a-tweet-form-needs-to-refresh-many-of-its-ui-elements-and-values)
* #42 - [Loosing tweets on threaded conversation.](http://20couch.lighthouseapp.com/projects/46779/tickets/42-loosing-tweets-on-threaded-conversation)
* #44 - [Don't open links in a new tab by default](http://20couch.lighthouseapp.com/projects/46779/tickets/44-dont-open-links-in-a-new-tab-by-default)
* #50 - [Duplicate items](http://20couch.lighthouseapp.com/projects/46779/tickets/50-duplicate-items)

Version 1.01 (March 14, 2010)
=============================
Features
--------
* Nada

Defects
-------
* Fixed an issue with a missing JavaScript file when doing auto-update.

Version 1.0 (March 14, 2010)
============================
Features
--------

app/config/core.php
<?php
/* SVN FILE: $Id: core.php 7690 2008-10-02 04:56:53Z nate $ */
/**
 * This is core configuration file.
 *
 * Use it to configure core behavior of Cake.
 *
 * PHP versions 4 and 5
 *
 * CakePHP(tm) :  Rapid Development Framework <http://www.cakephp.org/>
 * Copyright 2005-2008, Cake Software Foundation, Inc.
 *								1785 E. Sahara Avenue, Suite 490-204
 *								Las Vegas, Nevada 89104
 *
 * Licensed under The MIT License
 * Redistributions of files must retain the above copyright notice.
 *
 * @filesource
 * @copyright		Copyright 2005-2008, Cake Software Foundation, Inc.
 * @link				http://www.cakefoundation.org/projects/info/cakephp CakePHP(tm) Project
 * @package			cake
 * @subpackage		cake.app.config
 * @since			CakePHP(tm) v 0.2.9
 * @version			$Revision: 7690 $
 * @modifiedby		$LastChangedBy: nate $
 * @lastmodified	$Date: 2008-10-02 00:56:53 -0400 (Thu, 02 Oct 2008) $
 * @license			http://www.opensource.org/licenses/mit-license.php The MIT License
 */
/**
 * CakePHP Debug Level:
 *
 * Production Mode:
 * 	0: No error messages, errors, or warnings shown. Flash messages redirect.
 *
 * Development Mode:
 * 	1: Errors and warnings shown, model caches refreshed, flash messages halted.
 * 	2: As in 1, but also with full debug messages and SQL output.
 * 	3: As in 2, but also with full controller dump.
 *
 * In production mode, flash messages redirect after a time interval.
 * In development mode, you need to click the flash message to continue.
 */
	Configure::write('debug', 0);
/**
 * Application wide charset encoding
 */
	Configure::write('App.encoding', 'UTF-8');
/**
 * To configure CakePHP *not* to use mod_rewrite and to
 * use CakePHP pretty URLs, remove these .htaccess
 * files:
 *
 * /.htaccess
 * /app/.htaccess
 * /app/webroot/.htaccess
 *
 * And uncomment the App.baseUrl below:
 */
	//Configure::write('App.baseUrl', env('SCRIPT_NAME'));
/**
 * Uncomment the define below to use CakePHP admin routes.
 *
 * The value of the define determines the name of the route
 * and its associated controller actions:
 *
 * 'admin' 		-> admin_index() and /admin/controller/index
 * 'superuser' -> superuser_index() and /superuser/controller/index
 */
	//Configure::write('Routing.admin', 'admin');

/**
 * Turn off all caching application-wide.
 *
 */
	//Configure::write('Cache.disable', true);
/**
 * Enable cache checking.
 *
 * If set to true, for view caching you must still use the controller
 * var $cacheAction inside your controllers to define caching settings.
 * You can either set it controller-wide by setting var $cacheAction = true,
 * or in each action using $this->cacheAction = true.
 *
 */
	//Configure::write('Cache.check', true);
/**
 * Defines the default error type when using the log() function. Used for
 * differentiating error logging and debugging. Currently PHP supports LOG_DEBUG.
 */
	define('LOG_ERROR', 2);
/**
 * The preferred session handling method. Valid values:
 *
 * 'php'	 		Uses settings defined in your php.ini.
 * 'cake'		Saves session files in CakePHP's /tmp directory.
 * 'database'	Uses CakePHP's database sessions.
 *
 * To define a custom session handler, save it at /app/config/<name>.php.
 * Set the value of 'Session.save' to <name> to utilize it in CakePHP.
 *
 * To use database sessions, execute the SQL file found at /app/config/sql/sessions.sql.
 *
 */
	Configure::write('Session.save', 'php');
/**
 * The name of the table used to store CakePHP database sessions.
 *
 * 'Session.save' must be set to 'database' in order to utilize this constant.
 *
 * The table name set here should *not* include any table prefix defined elsewhere.
 */
	//Configure::write('Session.table', 'cake_sessions');
/**
 * The DATABASE_CONFIG::$var to use for database session handling.
 *
 * 'Session.save' must be set to 'database' in order to utilize this constant.
 */
	//Configure::write('Session.database', 'default');
/**
 * The name of CakePHP's session cookie.
 */
	Configure::write('Session.cookie', '20COUCH');
/**
 * Session time out time (in seconds).
 * Actual value depends on 'Security.level' setting.
 */
	Configure::write('Session.timeout', '12096');
/**
 * If set to false, sessions are not automatically started.
 */
	Configure::write('Session.start', true);
/**
 * When set to false, HTTP_USER_AGENT will not be checked
 * in the session
 */
	Configure::write('Session.checkAgent', false);
/**
 * The level of CakePHP security. The session timeout time defined
 * in 'Session.timeout' is multiplied according to the settings here.
 * Valid values:
 *
 * 'high'	Session timeout in 'Session.timeout' x 10
 * 'medium'	Session timeout in 'Session.timeout' x 100
 * 'low'		Session timeout in 'Session.timeout' x 300
 *
 * CakePHP session IDs are also regenerated between requests if
 * 'Security.level' is set to 'high'.
 */
	Configure::write('Security.level', 'medium');
/**
 * A random string used in security hashing methods.
 */
	Configure::write('Security.salt', '95e653c0e6d1e6bad756a31f10bbba8f2a2e3770');
/**
 * Compress CSS output by removing comments, whitespace, repeating tags, etc.
 * This requires a/var/cache directory to be writable by the web server for caching.
 * and /vendors/csspp/csspp.php
 *
 * To use, prefix the CSS link URL with '/ccss/' instead of '/css/' or use HtmlHelper::css().
 */
	//Configure::write('Asset.filter.css', 'css.php');
/**
 * Plug in your own custom JavaScript compressor by dropping a script in your webroot to handle the
 * output, and setting the config below to the name of the script.
 *
 * To use, prefix your JavaScript link URLs with '/cjs/' instead of '/js/' or use JavaScriptHelper::link().
 */
	//Configure::write('Asset.filter.js', 'custom_javascript_output_filter.php');
/**
 * The classname and database used in CakePHP's
 * access control lists.
 */
	Configure::write('Acl.classname', 'DbAcl');
	Configure::write('Acl.database', 'default');
/**
 *
 * Cache Engine Configuration
 * Default settings provided below
 *
 * File storage engine.
 *
 * 	 Cache::config('default', array(
 *		'engine' => 'File', //[required]
 *		'duration'=> 3600, //[optional]
 *		'probability'=> 100, //[optional]
 * 		'path' => CACHE, //[optional] use system tmp directory - remember to use absolute path
 * 		'prefix' => 'cake_', //[optional]  prefix every cache file with this string
 * 		'lock' => false, //[optional]  use file locking
 * 		'serialize' => true, [optional]
 *	));
 *
 *
 * APC (http://pecl.php.net/package/APC)
 *
 * 	 Cache::config('default', array(
 *		'engine' => 'Apc', //[required]
 *		'duration'=> 3600, //[optional]
 *		'probability'=> 100, //[optional]
 * 		'prefix' => Inflector::slug(APP_DIR) . '_', //[optional]  prefix every cache file with this string
 *	));
 *
 * Xcache (http://xcache.lighttpd.net/)
 *
 * 	 Cache::config('default', array(
 *		'engine' => 'Xcache', //[required]
 *		'duration'=> 3600, //[optional]
 *		'probability'=> 100, //[optional]
 * 		'prefix' => Inflector::slug(APP_DIR) . '_', //[optional] prefix every cache file with this string
 *		'user' => 'user', //user from xcache.admin.user settings
 *      'password' => 'password', //plaintext password (xcache.admin.pass)
 *	));
 *
 *
 * Memcache (http://www.danga.com/memcached/)
 *
 * 	 Cache::config('default', array(
 *		'engine' => 'Memcache', //[required]
 *		'duration'=> 3600, //[optional]
 *		'probability'=> 100, //[optional]
 * 		'prefix' => Inflector::slug(APP_DIR) . '_', //[optional]  prefix every cache file with this string
 * 		'servers' => array(
 * 			'127.0.0.1:11211' // localhost, default port 11211
 * 		), //[optional]
 * 		'compress' => false, // [optional] compress data in Memcache (slower, but uses less memory)
 *	));
 *
 */
	Cache::config('default', array('engine' => 'File'));
?>
